src/native/fs/fs_napi.cpp

```diff
@@ -1351,6 +1352,47 @@ struct GetPwName : public FSWorker
     }
 };
 
+/**
+ * Utimensat is an fs op
+ * change file times with nanosecond precision (access and modification times)
+ * see https://man7.org/linux/man-pages/man3/utimensat.3p.html
+ */
+struct Utimensat : public FSWorker
+{
+    std::string _path;
+    struct timespec _times[2];
+    Utimensat(const Napi::CallbackInfo& info)
+        : FSWorker(info)
+    {
+        _path = info[1].As<Napi::String>();
+        Napi::Object napi_times = info[2].As<Napi::Object>();
+        std::string new_atime;
+        get_time_from_info(napi_times, "actime", _times[0], new_atime);
+        std::string new_mtime;
+        get_time_from_info(napi_times, "modtime", _times[1], new_mtime);
+        Begin(XSTR() << "utimensat " << DVAL(_path) << DVAL(new_atime) << DVAL(new_mtime));
+    }
+    virtual void Work()
+    {
+        SYSCALL_OR_RETURN(utimensat(AT_FDCWD, _path.c_str(), _times, 0));
+    }
+
+   void get_time_from_info(const Napi::Object& napi_times, const std::string& time_type, struct timespec& timeval, std::string& log_str) {
+        if(napi_times.Get(time_type).IsBigInt()) {
+            const Napi::BigInt bigint = napi_times.Get(time_type).As<Napi::BigInt>();
+            //TODO: handle lossless
+            bool lossless = true;
+            const int64_t time = bigint.Int64Value(&lossless);
+            timeval.tv_sec = time / int64_t(1e9);
+            timeval.tv_nsec = time % int64_t(1e9);
+            log_str = std::to_string(time);
+        } else {
+            timeval.tv_nsec = UTIME_OMIT;
```

Comment:
when will this happen? was this flow tested?

Reply:
this will happen when one of the times is undefined. it means it will not be changed. it is tested in `Utimensat - change only mtime`

Reply:
but in namespaceFS we always take a defined mtime, I mean, I don't see how we can reach this code from our flow

Reply:
this function is used both by mtime and atime. currently we always get here for atime (I can also see this in the logs). also I think its good to leave an option to just change the atime and not the mtime, in case we want to use this function elsewhere

Reply:
got you, so for atime this flag will just keep it as it is since we pass only mtime